### ISSUE TITLE
Workflow.get_password() hangs since update to El Capitan

### ISSUE THREAD

**Issue by resamsel** [deanishe/alfred-workflow#71]
When calling the get_password method the script hangs and needs to be cancelled (CTRL-C). There's no feedback whatsoever, so I can't give you much more information than the traceback.

The code: https://github.com/resamsel/atlassian-alfred-workflow/blob/master/src/atlassian/config.py#L52

Trace after CTRL-C:
```
Traceback (most recent call last):
  File "/usr/local/bin/awf-config", line 9, in <module>
    load_entry_point('atlassian-alfred-workflow==0.0.1', 'console_scripts', 'awf-config')()
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/src/config.py", line 12, in main
    create_workflow().run(config.main)
  File "/usr/local/lib/python2.7/site-packages/Alfred_Workflow-1.13-py2.7.egg/workflow/workflow.py", line 2151, in run
    func(self)
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/src/atlassian/config.py", line 93, in main
    for issue in Config(wf).process():
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/src/atlassian/config.py", line 81, in process
    items = self.get_items(value)
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/src/atlassian/config.py", line 52, in get_items
    wf.get_password(KEY_PASSWORD),
  File "/usr/local/lib/python2.7/site-packages/Alfred_Workflow-1.13-py2.7.egg/workflow/workflow.py", line 2502, in get_password
    account, '-g')
  File "/usr/local/lib/python2.7/site-packages/Alfred_Workflow-1.13-py2.7.egg/workflow/workflow.py", line 2870, in _call_security
    retcode, output = p.wait(), p.stdout.read().strip().decode('utf-8')
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1384, in wait
    pid, sts = _eintr_retry_call(os.waitpid, self.pid, 0)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 476, in _eintr_retry_call
    return func(*args)
KeyboardInterrupt
```

**Comment by deanishe** [comment]
I can't replicate the problem. There are a couple of things you could try:

1. Run `security` yourself, e.g. `security get-generic-password -s <your.bundle.id> -a password -g`, to see whether it's an issue with `security`.
2. The way you're using Alfred-Workflow is unusual (you seem to be using it installed in `site-packages` and from an installed script). It will only run properly from within a workflow because it needs information from `info.plist` for a lot of its functionality (including Keychain). It looks for `info.plist` based on its own location and the current working directory (Alfred runs workflows from their own directories). That means the current working directory must be within a workflow or it needs to be installed in the workflow next to your other Python code (the preferred solution).

**Comment by resamsel** [comment]
Hi Dean,

thanks for the answer.

1. I was trying to use the command you offered, but did get an `security: unknown command "get-generic-password"` error. Did you mean find-generic-password? That one retrieves the stored password and displays it on the command line: `security find-generic-password -s <my.bundle.id> -a password -g`
2. I'm using a helper entry point to access the functionality of my Alfred workflow, which allows for debugging and cancelling the operation (which was not possible within the workflow). I'm within the directory with the plist file, so your library doesn't complain about not finding it ;)

Interestingly, it now works. What I did was remove all cached/saved data from the ~/Library/Application\ Support/Alfred\ 2/Workflow\ Data/<my.bundle.id> directory. Additionally, the command above (security ...) might have something to do with the solution.

As you cannot reproduce this issue and I'm not having that problem any longer I'll close this issue. Thanks for your help!

**Comment by resamsel** [comment]
The issue occurred again... This time I started the workflow in a virtualenv to simulate the way it works with Alfred. This is the output:

```
09:38:05 config2alfred.py:12 DEBUG    Args: []
09:38:05 config.py:42 DEBUG    Processing key jira-server
09:38:05 workflow.py:1667 DEBUG    Data `jira-server` stored in `cpickle` format
09:38:05 workflow.py:1682 DEBUG    Stored data loaded from : /Users/resamsel/Library/Application Support/Alfred 2/Workflow Data/com.resamsel.atlassian/jira-server.cpickle
09:38:05 config.py:42 DEBUG    Processing key bamboo-server
09:38:05 workflow.py:1667 DEBUG    Data `bamboo-server` stored in `cpickle` format
09:38:05 workflow.py:1682 DEBUG    Stored data loaded from : /Users/resamsel/Library/Application Support/Alfred 2/Workflow Data/com.resamsel.atlassian/bamboo-server.cpickle
09:38:05 config.py:42 DEBUG    Processing key username
09:38:05 workflow.py:1667 DEBUG    Data `username` stored in `cpickle` format
09:38:05 workflow.py:1682 DEBUG    Stored data loaded from : /Users/resamsel/Library/Application Support/Alfred 2/Workflow Data/com.resamsel.atlassian/username.cpickle
09:38:05 config.py:42 DEBUG    Processing key password
^C09:38:18 workflow.py:2176 DEBUG    Workflow finished in 13.064 seconds.
Traceback (most recent call last):
  File "config2alfred.py", line 21, in <module>
    sys.exit(create_workflow().run(main))
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/target/Atlassian Workflow/workflow/workflow.py", line 2151, in run
    func(self)
  File "config2alfred.py", line 14, in main
    for item in config.Config(wf).process():
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/target/Atlassian Workflow/atlassian/config.py", line 81, in process
    items = self.get_items(value)
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/target/Atlassian Workflow/atlassian/config.py", line 52, in get_items
    wf.get_password(KEY_PASSWORD),
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/target/Atlassian Workflow/workflow/workflow.py", line 2502, in get_password
    account, '-g')
  File "/Users/resamsel/Development/Python/atlassian-alfred-workflow/target/Atlassian Workflow/workflow/workflow.py", line 2870, in _call_security
    retcode, output = p.wait(), p.stdout.read().strip().decode('utf-8')
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1384, in wait
    pid, sts = _eintr_retry_call(os.waitpid, self.pid, 0)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 476, in _eintr_retry_call
    return func(*args)
KeyboardInterrupt
```

Last time it somehow worked after calling `security find-generic-password -s com.resamsel.atlassian -a password -g` (or maybe it's got nothing to do with it). I'm quite puzzled on why this sometimes works, and other times it doesn't...

**Comment by deanishe** [comment]
I'm afraid I don't see this as a bug with Alfred-Workflow. I can't replicate the issue. It's never been reported before.

Have you tried running `security` repeatedly from the command line to see if it ever hangs?

**Comment by deanishe** [comment]
This should be fixed now in #82. It appears that a full buffer was causing a deadlock.

I could never replicate the issue, but @fniephaus was having it, too, and says his PR fixes it.
